lib/pages/home_page.dart

```diff
@@ -0,0 +1,179 @@
+import 'package:todo_app/database/database.dart';
+import 'package:todo_app/util/add_task_dialogue_box.dart';
+import 'package:todo_app/util/todo_tile.dart';
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key});
+
+  @override
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  final _myBox = Hive.box('myBox');
+  final _controller = TextEditingController();
+  ToDoData db = ToDoData();
+  final String day = "monday";
+  final int dayIndex = 0;
+  @override
+  void initState() {
+    super.initState();
+    if(_myBox.get("TODOLIST") == null){
+      db.createInitialData();
+    }
+    else{
+      db.loadData();
+    }
+  }
+
+  void checkBoxChanged(bool? value, int index){
+    setState(() {
+      if(db.toDoList[day] != null){
+        db.toDoList[day]![index][1] = !db.toDoList[day]![index][1];
+      }
+    });
+    db.updateData();
+  }
+  void saveNewTask(){
+    setState(() {
+      if(db.toDoList[day] != null){
+        db.toDoList[day]!.add([_controller.text, false]);
+      }
+      if(db.isInitialTaskDone[dayIndex] == false && db.toDoList[day]!.length == 2 && db.toDoList[day]![0][0] == "Create a new task."){
+        db.isInitialTaskDone[dayIndex] = true;
+        db.toDoList[day]![0][1] = true;
+      }
+      _controller.clear();
+    });
+    // print(db.isInitialTaskDone[dayIndex]);
+    Navigator.of(context).pop();
+    db.updateData();
+  }
+
+  void addNewTask(){
+    showDialog(
+      context: context, 
+      builder: (context){
+        return DialogueBox(
+          controller: _controller,
+          onSave: saveNewTask,
+          onCancel: () {
+            _controller.clear();
+            Navigator.of(context).pop();
+          }
+        );
+      },
+    );
+  }
+
+  void deleteTask(int index){
+    setState(() {
+      if(db.toDoList[day] != null){
+        db.toDoList[day]!.removeAt(index);
+      }
+      _controller.clear();
+    });
+    db.updateData();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.yellow.shade200,
+      appBar: AppBar(
+        title: const Center(child: Text('TO DO FOR MONDAY')),
+        elevation: 0,
+      ),
+      drawer: Drawer(
+        child: ListView(
+          padding: EdgeInsets.zero,
+          children: <Widget>[
+            Container(
+              height: 150,
+              decoration: const BoxDecoration(
+                color: Colors.yellow,
+              ),
+              padding: const EdgeInsets.all(16.0),
+              alignment: Alignment.centerLeft,
+              child: const Text(
+                'Days',
+                style: TextStyle(fontSize: 24,),
+              )
+            ),
+            ListTile(
+              title: const Text('Monday'),
+              onTap: (){
+                Navigator.pop(context);
+                Navigator.pushNamed(context, '/monday');
```

Comment:
You can use pushReplacement to replace
https://api.flutter.dev/flutter/widgets/Navigator/pushReplacement.html

Reply:
everything done sir. 👍